File: roomnest (1)/roomnest/src/main/java/com/roomnest/entity/Room.java
package com.roomnest.entity;

import jakarta.persistence.*;
import lombok.*;

@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
@Entity
@Table(name = "rooms")
public class Room {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private String title;

    private Double rent;

    private Double deposit;

    private String city;

    private String area;

    private String address;

    private String roomType;

    private String gender;

    @Column(length = 1000)
    private String description;

    private boolean available = true;

    @ManyToOne
    @JoinColumn(name = "owner_id")
    private User owner;
}

File: Frontend/lib/models/room.dart
class Room {
  final int id;
  final String title;
  final double rent;
  final double deposit;
  final String city;
  final String area;
  final String address;
  final String roomType;
  final String gender;
  final String description;
  final bool available;

  Room({
    required this.id,
    required this.title,
    required this.rent,
    required this.deposit,
    required this.city,
    required this.area,
    required this.address,
    required this.roomType,
    required this.gender,
    required this.description,
    required this.available,
  });

  factory Room.fromJson(Map<String, dynamic> json) {
    return Room(
      id: json["id"],
      title: json["title"] ?? "",
      rent: (json["rent"] as num).toDouble(),
      deposit: (json["deposit"] as num).toDouble(),
      city: json["city"] ?? "",
      area: json["area"] ?? "",
      address: json["address"] ?? "",
      roomType: json["roomType"] ?? "",
      gender: json["gender"] ?? "",
      description: json["description"] ?? "",
      available: json["available"] ?? false,
    );
  }

  Map<String, dynamic> toJson() {
    return {
      "id": id,
      "title": title,
      "rent": rent,
      "deposit": deposit,
      "city": city,
      "area": area,
      "address": address,
      "roomType": roomType,
      "gender": gender,
      "description": description,
      "available": available,
    };
  }
}
